Feature/pc console overview

zbialecki opened the pull request:
Updates Configuration tab to display a variety of useful statistics on the overall progress of the conference.

Note: when updating an existing PC console webfield using the group editor, make sure to add `WITHDRAWN_SUBMISSION_ID` and `DESK_REJECTED_SUBMISSION_ID` to the constants section.

Fixes #598
Fixes #563 
Fixes https://github.com/openreview/openreview/issues/1213
Fixes #408

melisabok commented:
Currently ECCV status:

<img width="1263" alt="Screen Shot 2020-03-23 at 5 37 07 PM" src="https://user-images.githubusercontent.com/545506/77365725-000c8d80-6d2d-11ea-925e-3fcc1fa1d313.png">


I really like this change, it gives more information about the status of the conference. 

All the numbers are associated to an invitation. Another approach is too a show the status of each invitation independently building like a workflow. Like:

* Recruitment: invited reviewers, reviewers, ac,
* Submission: active submissions, desk reject, withdraw
* Bid: status of completion
* Recommendation: status of completion.

All the invitations with their start, due and exp date. Render differently the ones that are already expired, active or be active in the future. 

About the Review stage, I would show the number of reviews submitted, sometimes it takes time to complete all the reviews for a specific reviewers. 

# All Reviews
# Papers with reviews completed.
# Reviewers with reviews completed.

zbialecki commented:
OK, I can improve the stats on reviews to include the 3 bullets that you suggested. 

For recruitment stats, are you suggesting adding a new row of stats at the top for # reviewers invited/accepted, and # ACs invited/accepted? Or are you suggesting a different layout for all of these stats? How would I get the number of reviewers invited?

melisabok commented:
I'm suggesting to change the layout where we can group the data per invitation/stage. But this is a suggestion! I'm not sure we should experiment with different layouts. 

With this version we are showing the numbers/stats at the top and the timeline at the bottom, where they are highly related. 

Invited/Declined are groups where you can check the members length. You can see the links in Venue Roles.

zbialecki commented:
I added the stats you suggested and changed the layout slightly to highlight the different stages more. Here's what it looks like with some out-of-date ICLR data:

<img width="1164" alt="Screen Shot 2020-03-26 at 11 35 30 PM" src="https://user-images.githubusercontent.com/86174/77719096-d10b4b80-6fba-11ea-877d-612501d307c8.png">

mohituniyal commented:
I love these changes @zbialecki :)

melisabok commented:
It looks better!

I still think we should try to show stats and timeline all together group by invitation/stage. And highlight only the stage that is active otherwise you can get lost with too many numbers. It should be one line per invitation, I guess we should use a sketch tool to try this. 

There is a typo in Paper progress: '% of papers that have reveived reviews from all assigned reviewers'. 

In 'Review Submitted' we could know how many reviews we expect to have. I guess counting the number of AnonReviewer groups.

zbialecki commented:
How do you get the current stage of a venue? Also, what do you mean by 1 line per invitation?

melisabok commented:
Checking the cdate, duedate and expdate of each invitation. If now is between cdate and expdate then the stage is active. 

I think something like the timeline, where we have one invitation per line plus adding the stats next to each line. Again, I'm not very sure about this, I like your version. We could ask the PCs for more feedback.

zbialecki commented:
OK, I think I'm starting to understand what you're describing. I do think that would be cool, but I think there would have to be multiple invitations per stage/row of stats. For example, the recruitment stage has Recruit_Reviewers and Recruit_Area_Chairs. I think getting some feedback from actual PCs at this stage would be very helpful.

melisabok commented:
Some feedback:

- I guess this is happening because is diving by zero:

<img width="376" alt="Screen Shot 2020-03-31 at 9 43 19 AM" src="https://user-images.githubusercontent.com/545506/78034297-8892bb80-7335-11ea-9aa8-1542f2465a65.png">

- Can we show the invitation description in a single line, may be reduce the timezone string?

<img width="775" alt="Screen Shot 2020-03-31 at 9 43 31 AM" src="https://user-images.githubusercontent.com/545506/78034348-9ba58b80-7335-11ea-9306-0c8f8f40aee5.png">

- I'm getting an error when I try to assign a reviewer to a paper:

<img width="1802" alt="Screen Shot 2020-03-31 at 9 45 19 AM" src="https://user-images.githubusercontent.com/545506/78034409-b2e47900-7335-11ea-98d9-3a2252c6a61b.png">

- I'm also noticing that `SHOW_AC_TAB` is not being used anymore. This variable specifies if the venue has area chairs or not. I think we should keep using it to avoid extra network calls and hide the AC status tab.

zbialecki commented:
Thanks for testing Melisa. Those issues are now fixed. About `SHOW_AC_TAB`: that constant is now used to turn off the AC tab and some data specific to that tab, but to truly stop all AC data from being loaded it's necessary to set the `AREA_CHAIRS_ID` constant to be empty.

melisabok commented:
So, let's remove `SHOW_AC_TAB` constant and use `AREA_CHAIRS_ID` to show/hide the tab.

melisabok commented:
This is great!!!

Now that I remember Reviewers and ACs have reminder functionality, why do we have reminders in the Paper Status tab?

![Screen Shot 2020-04-01 at 1 26 29 PM](https://user-images.githubusercontent.com/545506/78167556-b141b080-741c-11ea-9d6d-73fd35927e3c.png)

This is the option that sends one message request per reviewer/paper. Should we remove it?

zbialecki commented:
That option actually sends one request per paper (could be multiple reviewers), so it does group the requests when it can. I would prefer to keep it because that has some nice features like allowing you to select individual papers to message, and also includes the review link in the email.

melisabok commented:
ok, let's keep it. It can be confusing for the PC too many options to remind reviewers. 

I found a bug in the reminder, we should remind reviewers/acs that have assigned papers and missing reviews/meta-reviews:

![Screen Shot 2020-04-01 at 1 43 55 PM](https://user-images.githubusercontent.com/545506/78169288-257d5380-741f-11ea-9f10-d735e48b2a23.png)

Looking at the screenshot we should remind to 'Reviewer ECCV_One' and 'Reviewer ECCV_Two'.

zbialecki commented:
That should be fixed now, let me know if you see any other issues.

melisabok commented on openreview/conference/templates/programchairWebfield.js:
```diff
@@ -1453,10 +1453,10 @@ var displayAreaChairsStatusTable = function() {
       },
       'msg-unsubmitted-reviews': function(row) {
-        return row.reviewProgressData.numCompletedReviews === 0;
+        return row.reviewProgressData.numCompletedReviews === 0 && row.reviewProgressData.numPapers > 0;
```
I think this should be:

```
return row.reviewProgressData.numCompletedReviews < row.reviewProgressData.numPapers
```

A reviewer can have 2 papers assigned and entered one review and we should remind them.

zbialecki replied:
Ah you're totally right
